src/model/ElectricLoadCenterStorageLiIonNMCBattery.hpp

```diff
@@ -0,0 +1,194 @@
+*  following conditions are met:
+*
+*  (1) Redistributions of source code must retain the above copyright notice, this list of conditions and the following
+*  disclaimer.
+*
+*  (2) Redistributions in binary form must reproduce the above copyright notice, this list of conditions and the following
+*  disclaimer in the documentation and/or other materials provided with the distribution.
+*
+*  (3) Neither the name of the copyright holder nor the names of any contributors may be used to endorse or promote products
+*  derived from this software without specific prior written permission from the respective party.
+*
+*  (4) Other than as required in clauses (1) and (2), distributions in any form of modifications or other derivative works
+*  may not use the "OpenStudio" trademark, "OS", "os", or any other confusingly similar designation without specific prior
+*  written permission from Alliance for Sustainable Energy, LLC.
+*
+*  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDER(S) AND ANY CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES,
+*  INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+*  DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER(S), ANY CONTRIBUTORS, THE UNITED STATES GOVERNMENT, OR THE UNITED
+*  STATES DEPARTMENT OF ENERGY, NOR ANY OF THEIR EMPLOYEES, BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+*  EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF
+*  USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+*  STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF
+*  ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+***********************************************************************************************************************/
+
+#ifndef MODEL_ELECTRICLOADCENTERSTORAGELIIONNMCBATTERY_HPP
+#define MODEL_ELECTRICLOADCENTERSTORAGELIIONNMCBATTERY_HPP
+
+#include "ModelAPI.hpp"
+#include "ElectricalStorage.hpp"
+
+namespace openstudio {
+
+namespace model {
+
+  class Schedule;
+  class ThermalZone;
+
+  namespace detail {
+
+    class ElectricLoadCenterStorageLiIonNMCBattery_Impl;
+
+  }  // namespace detail
+
+  /** ElectricLoadCenterStorageLiIonNMCBattery is a ElectricalStorage that wraps the OpenStudio IDD object 'OS:ElectricLoadCenter:Storage:LiIonNMCBattery'. */
+  class MODEL_API ElectricLoadCenterStorageLiIonNMCBattery : public ElectricalStorage
+  {
+   public:
+    /** @name Constructors and Destructors */
+    //@{
+
+    explicit ElectricLoadCenterStorageLiIonNMCBattery(const Model& model, const int numberofCellsinSeries, const int numberofStringsinParallel,
+                                                      const double batteryMass, const double batterySurfaceArea);
```

Comment:
I changed the constructor to require these additional fields.

Reply:
I'm not a big fan of this change. If we can provide reasonable defaults (I think we can, just pick whatever is your the E+ example file) I'd prefer to have a `ElectricLoadCenterStorageLiIonNMCBattery(const Model& model)` ctor.

You can have 2 constructors if you want this explicit one. And you could still throw in the ReverseTranslator if the IDF doesn't have one of these fields if you like (or issue a Warn/Error, and let the default pass in)

(This isn't "out the blue", we've had lengthy discussions about this in the past).